fs/base.py

```diff
@@ -1645,3 +1662,8 @@ def hash(self, path, name):
                     break
                 hash_object.update(chunk)
         return hash_object.hexdigest()
+
+
+def has_drive_letter(path_snippet):
```

Comment:
Could we put this at the top of the file and make it private (I'm assuming its not used outside of this file).

Docstring and typing comments please...

Reply:
added.